Add pagination when listing users of a team

guewen opened the pull request:
Allows to have more than 30 users in a team.

Fixes #11

Commit:
Add pagination when listing users of a team

Allows to have more than 30 users in a team.

Fixes #11

raphink commented on main.go:
```diff
@@ -130,8 +130,21 @@ func (p *gitHubPki) getTeamUsers() (err error) {
+					checkErr(err, "Failed to list team members for team "+*team.Name+": %v")
+					for _, u := range pageUsers {
+						ghUsers = append(ghUsers, *u)
+					}
```
iirc `ghUsers = append(ghUsers, pageUsers...)` would work

raphink commented on main.go:
```diff
@@ -130,8 +130,21 @@ func (p *gitHubPki) getTeamUsers() (err error) {
+					if resp.NextPage == 0 {
+						break
+					}
+					opt.Page = resp.NextPage
```
iirc you could write

```go
if opt.Page = resp.NextPage; opt.Page == 0 {
  break
}
```

Commit:
fixup! Add pagination when listing users of a team

guewen commented:
@raphink Thanks! I confirm both your changes work, added a fixup commit

raphink commented on main.go:
```diff
@@ -130,8 +130,18 @@ func (p *gitHubPki) getTeamUsers() (err error) {
-				checkErr(err, "Failed to list team members for team "+*team.Name+": %v")
+				var ghUsers []*github.User
+				opt := &github.TeamListTeamMembersOptions{
+					ListOptions: github.ListOptions{PerPage: 100},
```
If we're following pagination, do we really need to increase the amount of results per page?

guewen replied:
I used the same value as the listing of teams done above, but you are right, it's probably better to keep the default value. BTW, do you want me to update the listing of teams with the same changes as I did there to be consistent?

raphink replied:
Sure, good idea

guewen replied:
Done

Commit:
fixup! fixup! Add pagination when listing users of a team

raphink commented on main.go:
```diff
@@ -130,8 +130,18 @@ func (p *gitHubPki) getTeamUsers() (err error) {
+					ListOptions: github.ListOptions{},
+				}
+				for {
+					pageUsers, resp, err := p.Client.Teams.ListTeamMembers(context.Background(), *team.ID, opt)
```
I think `opt` could remain `nil` as it was before since we're not overriding anything.

guewen replied:
We have to modify the `Page` of the `opt` object for the calls after the first page.

Commit:
Modify the code for listing teams with pagination

To be consistent with the recent changes done in listing the users of a
team:

* We do not really need to change the number of records per page
* Simplify the pagination idiom, adopting the proposition on https://github.com/google/go-github#pagination
  and @raphink suggestions

Functionally, there's no difference at the end.
